ProcessActors.py

```diff
@@ -360,15 +583,125 @@ def process_pot(actor_bytes):
+        return {
+            "type": "Bush Circle",
+            "drop_table": drop_table
+        }
+    elif type == 1: # Scattered shrubs, random drops
+        return {
+            "type": "Scattered Bushes",
+            "drop_table": drop_table
+        }
+    else: # Rocks/don't care
+        pass
+def get_overlay_table_entry(rom, actor_id):
+    base = 0x00B5E490 + actor_id * 0x20
+    vrom_start = rom.read_int32(base)
+    vrom_end = rom.read_int32(base + 4)
+    vram_start = rom.read_int32(base + 8)
+    vram_end = rom.read_int32(base + 12)
+    file_ram = rom.read_int32(base + 16)
+    init_addr = rom.read_int32(base + 20)
+
+    if vram_start:
+        init_vrom = init_addr - vram_start + vrom_start
+    else:
+        init_vrom = 0xA87000 + init_addr - 0x800110A0
+
+    init_object = rom.read_int16(init_vrom + 8)
+    return {
+        'vrom_start': vrom_start,
+        'vrom_end': vrom_end,
+        'vram_start': vram_start,
+        'vram_end': vram_end,
+        'file_ram': file_ram,
+        'init_addr': init_addr,
+        'init_object':init_object
+    }
+
+
+def get_bad_actors(rom: Rom, scenes_data: list[Scene]):
+    bad_actors = []
+    keep_list = [1,2,3]
+    for scene in scenes_data:
+        for room in scene.rooms:
+            for setup in room.setups:
+                i = 0
+                for actor in room.setups[setup].actors:
+                    if actor.id != 0xFFFF: # Ignore any actors we've already patched out
+                        overlay_entry = get_overlay_table_entry(rom, actor.id) # Read the entry from the overlay table
+                        init_obj = overlay_entry['init_object']
+                        # Pots are dumb so handle them seperately
+                        if actor.id == 0x111:
+                            if scene.keep_id == 0x0003: # Dungeon pot
+                                init_obj = 0x0003
+                            else:
+                                init_obj = 0x12C # Overworld pots use their own object
+
+                        # ignore ganons tower rubble
+                        if actor.id == 0x0174:
+                            continue
+
+                        # Grass is also dumb
+                        if actor.id == 0x0125:
+                            params = rom.read_int16(actor.addr + 14)
+                            grass_obj_ids = [2, 0x012B, 0x012B]
+                            init_obj = grass_obj_ids[params & 0x0003]
+
+                        if init_obj not in room.setups[setup].objects and init_obj != scene.keep_id and init_obj != 1 and init_obj <= 0x0191: # Check if the init object is in the rooms data
+                            bad_actors.append((actor.addr, scenes[scene.id], scene.id, room.id, setup, i, actor.id, overlay_entry['init_object']))
+                        i += 1
+    for actor in bad_actors:
+        print(actor)
+    return bad_actors
+
+
+#rom = Rom("ZOOTDEC.z64")
```

Comment:
what's with all the dead code here

Reply:
This is a utility script from @rrealmuto used on his branch. Some of this was already in the main repo. I can remove the whole file if needed and its not inconvenient to Rob.

Reply:
Rob says it's fine to remove it.